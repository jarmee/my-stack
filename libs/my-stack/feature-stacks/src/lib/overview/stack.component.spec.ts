import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { StackComponent } from './stack.component';

const CSS_SELECTORS = {
  DELETE_BUTTON: By.css('[data-test-id="delete-button"]'),
};

describe('StackComponent', () => {
  let component: StackComponent;
  let fixture: ComponentFixture<StackComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [StackComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(StackComponent);
    component = fixture.componentInstance;
    component.stack = {
      id: 0,
      title: 'test-0',
      technologies: [],
    };
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('delete stack', () => {
    it('should emit a delete event for the stack when the user clicks the delete button', () => {
      component.delete.emit = jest.fn();

      fixture.debugElement
        .query(CSS_SELECTORS.DELETE_BUTTON)
        .triggerEventHandler('click');

      expect(component.delete.emit).toHaveBeenCalledWith(component.stack);
    });
  });
});
