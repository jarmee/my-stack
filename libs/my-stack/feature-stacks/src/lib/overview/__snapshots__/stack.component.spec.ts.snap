// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackComponent should match snapshot 1`] = `
<mys-stack
  __ngContext__={[Function Number]}
  delete={[Function EventEmitter_]}
  stack={[Function Object]}
>
  <div
    class="flex flex-col items-center justify-items-center cursor-pointer"
  >
    <mat-card
      class="mat-mdc-card mdc-card h-[144px] w-[144px]"
    >
      <button
        class="mdc-icon-button mat-mdc-icon-button mat-unthemed mat-mdc-button-base"
        data-test-id="delete-button"
        mat-icon-button=""
        type="button"
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
        />
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
          delete
        </mat-icon>
        <span
          class="mat-mdc-focus-indicator"
        />
        <span
          class="mat-ripple mat-mdc-button-ripple"
          matripple=""
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </button>
    </mat-card>
    <span
      class="mt-2"
    >
      test-0
    </span>
  </div>
</mys-stack>
`;
